### ISSUE TITLE
WebdriverIOAsync findElement(s) signatures have WebdriverIO return types

### ISSUE THREAD

**Issue by richsilv** [webdriverio/webdriverio#3760]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.7.5
 - **Mode:** Standalone

**Describe the bug**

The use of `WrapWithPromise` to generate `WebdriverIOAsync` types from `WebdriverIO` types means that the return types of most methods are correctly typed as Promises rather than the resolved types of those Promises.

However, in the case of `WebdriverIOAsync.Browser.findElement(s)` (or `$` or `$$`), the resolved type is itself an element, but since these method signatures have been generated from `WebdriverIO.Browser`, that element is of type `WebdriverIO.Element`, *not* `WebdriverIOAsync.Element` as it should be.

**To Reproduce**
```
type T = ReturnType<WebdriverIOAsync.Browser["$"]>; // Promise<WebdriverIO.Element>
```

**Expected behavior**
```
type T = ReturnType<WebdriverIOAsync.Browser["$"]>; // Promise<WebdriverIOAsync.Element>
```

So that we can do the following without a type error: 

```
const element = await browser.$("selector");
await element.$("sub-selector").then((subElement) => subElement.click());
```

**Comment by richsilv** [comment]
Fixes in #3768

**Comment by christian-bromann** [comment]
Let's continue convo there.
